sematic/config/settings.py

```diff
@@ -157,7 +157,11 @@ def get_plugin_settings(
     if plugin_path not in settings:
         raise MissingSettingsError(plugin)
 
-    return settings[plugin_path]
+    return {
+        key: value
+        for key, value in settings[plugin_path].items()
+        if key != _PLUGIN_VERSION_KEY
```

Comment:
This might need to be done in other places we have missed as well. Also, this code block looks like it pertains to the `PluginSettings` itself, and if here due to necessity.

In order to do this change in a sustainable way, how about changing the definition from:
```python
PluginSettings = Dict[Union[AbstractPluginSettingsVar, Literal["__version__"]], str]
```
to:
```python
class PluginSettings(Dict[Union[AbstractPluginSettingsVar, Literal["__version__"]], str]):
  [...]
  def get_settings(self):
    return {
      key: value
      for key, value in self.items()
      if key != _PLUGIN_VERSION_KEY
    }
  [...]
```

Reply:
This is actually not an issue since this is the only "official" API to obtain settings for a plug-in. All other ways are "illegal".